ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceAppliedChanges.h

```diff
@@ -32,4 +32,6 @@
                        insertedIndexPaths:(NSSet *)insertedIndexPaths
                                  userInfo:(NSDictionary *)userInfo NS_DESIGNATED_INITIALIZER;
 
+- (NSIndexPath *)newIndexPathForPreviousIndexPath:(NSIndexPath *)indexPath;
```

Comment:
Maybe something like this:

```
NSDictionary<NSIndexPath *, NSIndexPath *> *CKComputeFinalUpdatedIndexPathForAppliedChanges(CKTransactionalComponentDataSourceAppliedChanges *appliedChanges);
```

The result can be a dictionary from original updated index paths to the final updated index paths.